packages/components/src/components/pin-code/pin-code-utils.spec.ts

```diff
@@ -0,0 +1,243 @@
+        '::slotted(input:not(:disabled):not(:focus):not([readonly]):hover)': {
+          border: '2px',
+        },
+      },
+    };
+
+    const stylesWithoutSlottedSelector = {
+      input: {
+        display: 'none',
+      },
+      '@media(hover:hover)': {
+        'input:not(:disabled):not(:focus):not([readonly]):hover': {
+          border: '2px',
+        },
+      },
+    };
+
+    expect(removeSlottedSelector(stylesWithSlottedSelector)).toStrictEqual(stylesWithoutSlottedSelector);
+  });
+});
+
+describe('removeHoverStyles()', () => {
+  it('should remove @media(hover:hover) styles from Styles object', () => {
+    const stylesWithMediaHover = {
+      label: {
+        display: 'block',
+        '@media(hover:hover)': {
+          '::slotted(input:not(:disabled):not(:focus):not([readonly]):hover)': {
+            border: '2px',
+          },
+        },
+      },
+      input: {
+        display: 'none',
+      },
+      '@media(hover:hover)': {
+        '::slotted(input:not(:disabled):not(:focus):not([readonly]):hover)': {
+          border: '2px',
+        },
+      },
+    };
+
+    const stylesWithoutMediaHover = {
+      label: {
+        display: 'block',
+      },
+      input: {
+        display: 'none',
+      },
+    };
+
+    expect(removeHoverStyles(stylesWithMediaHover)).toEqual(stylesWithoutMediaHover);
+  });
+});
+
+describe('warnIfInitialValueIsTransformed()', () => {
+  it('should call getTagNameWithoutPrefix() and consoleWarn() with correct parameters', () => {
+    const host = document.createElement('p-pin-code');
+    const warningPrefix = `Property value on component p-pin-code:`;
+    const spyGetTagNameWithoutPrefix = jest.spyOn(getTagNameWithoutPrefixUtils, 'getTagNameWithoutPrefix');
+    const spyConsoleWarn = jest.spyOn(consoleWarnUtils, 'consoleWarn').mockImplementation();
+
+    warnIfInitialValueIsTransformed(host, 4);
+
+    expect(spyGetTagNameWithoutPrefix).toBeCalledTimes(1);
+    expect(spyGetTagNameWithoutPrefix).toBeCalledWith(host);
+    expect(spyConsoleWarn).toBeCalledWith(
+      warningPrefix,
+      'Provided value has too many characters and was truncated to the max length of 4.'
+    );
+
+    warnIfInitialValueIsTransformed(host);
+
+    expect(spyGetTagNameWithoutPrefix).toBeCalledTimes(2);
+    expect(spyGetTagNameWithoutPrefix).toBeCalledWith(host);
+    expect(spyConsoleWarn).toBeCalledWith(
+      warningPrefix,
+      'Provided value contains characters that are not of type number, the value was therefore reset.'
+    );
+  });
+});
+
+describe('isInputSingleDigit()', () => {
+  it.each<[string]>([['12'], ['abc'], ['a'], ['/^'], ['^']])('should return false for value: %s', (value) => {
+    expect(isInputSingleDigit(value)).toBeFalsy();
+  });
+
+  it('should return true if the provided string is a single digit', () => {
+    const spy = jest.spyOn(pinCodeUtils, 'isInputSingleDigit');
+
+    isInputSingleDigit('1');
+    expect(spy).toReturnWith(true);
```

Comment:
Usually in these scenarios we store the result in a variable and compare its value.